### DISCUSSION TITLE
【项目经验总结】

### DISCUSSION THREAD

**Discussion by swustzzh** [swustzzh/coder_books#26] [General]
1. 背景和目标
- 深入理解、分析、宣贯到位
- 方向、思路、方案和结果的基础，有指导作用
- 每一次需求评审、方案设计、问题解决融汇贯通到整个产品、业务、逻辑的理解和设计中，提前发现隐藏问题，降低需求变更、设计脱节、实现返工的概率
- 培养、提升认知、理解能力的关键因素
2. 设计
- 做设计或任务拆分时，视版本迭代节奏控制设计方式
- 结合项目实际迭代情况，注意总结、培养团队抽象能力
  - 没时间做codereview?可以在问题对应的代码是做codereview并在组内宣贯
  - 逐步抽象，慢慢改进；逐步升级优化，避免大范围改造花费巨大成本
- 先做初稿，提前熟悉，节省讨论时间，便于完善细节。
3. 问题处理
- 宣贯背景和目标的重要性，形成思考根本问题的习惯，提高理解能力
- 拉齐PM、FE、BE、QA，及时记录
- 组织内部复盘相关问题，解决主要问题，不要形式化
- 出现问题，及时分析和处理，解决问题，不针对某个人
- 团队气氛出现懈怠、懒惰、情绪化问题是，及时组织会议沟通，该强硬强硬，该心软心软
4. 环境
- 尽量配齐开发、测试、预发、生产四套环境
- 多版本开发时，可以利用开发、测试、语法对多个并行版本进行控制
- 不允许变更生产的任何配置，任何生产环境的变化必须经过测试、预发验证，绝不允许开发直接上生产
5. 权限
- 绝对控制生产环境的访问权限和操作动作
- 运维和研发负责人有访问权限，但非紧急情况下研发负责人应配合运维执行生产操作动作
- 有工具平台后，只有运维能访问生产环境，项目组其他所有成员只能走工具平台
- 生产系统开放给开发、测试的访问权限也要在一定可控范围内，比如操作日志留痕、系统日志打印用户、研发负责人对其他用户的权限管理等
6. 流程
- 上线流程从提测到预发到生产，有工具平台前，形式可以后补，但实际的上线操作流程必须要有必须严格执行，不能跳过配置验证、sql验证、各环境测试等环节
-   **任何时间、任何版本不允许未经测试人员验收直接上生产**
-   研发负责人全程跟踪上线流程直到生产验收完成（或特殊情况下由前后端负责人跟踪）
7. 工具辅助
- CI/CD
- 监控告警
- 在线生产日志
- 在线生产db审批
- 后期可以把这些集成到一起
8. 项目会议
- 内部例会发挥应有作用，快速同步进度、重点关注问题、及时传达事项、明确当天目标、带动工作节奏，不要流于形式
- 例会决议的方案、行动点、责任人等内容，后续按步骤要跟踪进度，让团队成员养成习惯--明白会议讨论的东西不是说说就算了，可以有效提升会议决策质量，还能让例会起到提醒大家快速进入工作状态的作用
- 注意把控节奏

**Comment by swlws** [comment]
团队管理方法：

- 好的方式
  - 交叉测试。
  - 适当分配任务。由易到难分配任务，提高自信
  - 总结分享。以项目中使用到的技术作为分享点，提高个人声望、以及团队建设
  - 踩坑经验。留存为团队知识库
- 不好的方式
  - 排行榜。易造成内卷，疲惫
